README.md

```diff
@@ -2,79 +2,118 @@
+repo to public github or specific customers.
+
+### Project composition / Workspaces
+
+Simplify code sharing and dependency management. Beyond just subdirectories,
+Josh supports selecting, re-mapping and compsing of arbitary virtual repositories
+from the content found in the monorepo.
+
+The mapping itself is also stored in the repository and therefore versioned alongside
+the code.
+
+<table>
+    <thead>
+        <tr>
+            <th>Central monorepo</th>
+            <th>Project workspaces</th>
+            <th>workspace.josh file</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td rowspan=2><img src="docs/src/img/central.svg?sanitize=true" alt="Folders and files in central.git" /></td>
+            <td><img src="docs/src/img/project1.svg?sanitize=true" alt="Folders and files in project1.git" /></td>
+            <td>
+<pre>
+dependencies = :/modules:[
+    ::tools/
+    ::library1/
+]
+</pre>
+        </tr>
+        <tr>
+            <td><img src="docs/src/img/project2.svg?sanitize=true" alt="Folders and files in project2.git" /></td>
+            <td>
+<pre>libs/library1 = :/modules/library1</pre></td>
+        </tr>
+    </tbody>
+</table>
+
+Workspaces act as normal git repos:
+
+```
+$ git clone http://josh/central.git:workspace=workspaces/project1.git
+```
+
+### Simplfied CI/CD
+
+With everything stored in one repo, CI/CD system only need to look into one source for each particular
+deliverable.
+However, building multiple deliverables from from one big tree, introduces a new complexity
+into the build system: Now build tools or package managers, typically taylored to specfic languages
+or toolchains, need to be used to answer the question: "What deliverables are affected by a given commit
+and need to be rebuild?".
+
+With Josh, each deliverable gets it's own virtual git repository with dependencies declared in the `workspace.josh`
+file. This means answering above question becomes as simple as comparing commit ids.
+Furthermore due to the tree filtering each build is guaranteed to be perfectly sandboxed
+and only sees those parts of the monorepo that have actually been mapped.
+
+This also means the deliverables to be re-build can be selected without cloning any repos like
+typically necessary with normal build tools.
+
+### GraphQL API
+
+It is often desireable to access content stored in git without requireing a clone of the repository.
+This is usefull for CI/CD systems or web frontends like dashboards.
+
+Josh exposes a GraphQL API for that purpose. For example it can be used to find all workspaces currently
+present in the tree:
+
+```
+query {
+  rev(at:"refs/heads/master", filter:"::**/workspace.josh") {
+    files { path }
+  }
+}
+```
+
+
+### Caching proxy
+
+Even without using the more advanced features like partial cloning or workspace,
+`josh-proxy` can act as a cache to reduce traffic between locations or keep your CI from
+doing lot's of requests to the main git host.
```

Comment:
doing -> performing
I'd rather you chose "many" over "lots of", so long as you also change "doing" to "performing"